### PR TITLE
7.0 rabobank reworked therp

### PR DESCRIPTION
**Pull request by NL66278**
Import for Rabobank MT940 was not working in most cases. This PR includes an earlier pull request to get the ING MT940 working. Furthermore added some more debugging for invalid transactions or statements.

### CODE REVIEW

**Comment by NL66278** [comment]
Supersedes: https://github.com/OCA/bank-payment/pull/83

File: account_banking_nl_rabo_mt940/account_banking_nl_rabo_mt940.py [hunk 1]
```diff
@@ -42,85 +39,57 @@ class RaboMT940Parser(MT940, parser):
...
+        # The alternative would be to split the file and start a new
+        # statement for each 20: tag encountered.
+        stmt = self.current_statement
+        if not stmt.local_currency:
+            stmt.local_currency = data[7:10]
+            stmt.date = datetime.strptime(data[1:7], '%y%m%d')
+            stmt.start_balance = str2amount(data[0], data[10:])
+            stmt.id = '%s-%s' % (
+                self.current_statement.date.strftime('%Y-%m-%d'),
+                self.current_statement.id)
```
Comment (StefanRijnhart) [inline account_banking_nl_rabo_mt940/account_banking_nl_rabo_mt940.py]: You are replacing a method to guarantee a unique identifier with something more static. Is this guaranteed to be unique?

File: account_banking_nl_rabo_mt940/account_banking_nl_rabo_mt940.py [hunk 2]
```diff
@@ -1,8 +1,9 @@
 # -*- coding: utf-8 -*-
+"""Implement parser for MT940 files - Rabobank dialect."""
 ##############################################################################
 #
 #    OpenERP, Open Source Management Solution
-#    This module copyright (C) 2014 Therp BV (<http://therp.nl>).
+#    Copyright (C) 2015 Therp BV <http://therp.nl>.
```
Comment (StefanRijnhart) [inline account_banking_nl_rabo_mt940/account_banking_nl_rabo_mt940.py]: Next time don't remove the year but can add the current year, e.g.

```
Copyright (C) 2014-2015
```

Just leave it for now

**Comment by StefanRijnhart** [comment]
Thanks, nice cleanup. Flake8  wants an extra line somewhere and Pylint says there are two print statements.

Curious to hear from @hbrunn as he has strong knowledge of the mt940 format.

**Comment by yvaucher** [comment]
Your 2 first commits are missing the email can you correct this with a rebase and commit --amend ?

File: account_banking_nl_rabo_mt940/account_banking_nl_rabo_mt940.py (hunk 1 above)
Comment (NL66278) [inline account_banking_nl_rabo_mt940/account_banking_nl_rabo_mt940.py]: The original method get_unique_account_identifier will return the same identifier for the same account in all cases. Maybe it was intended to call get_unique_statement_id instead? But in both cases the cursor is needed to query the database. I think the parser should just use the information in the parsed file itself. 
If there are problems with uniqueness, it should be solved outside of the parser. But I did not see one so far.

Comment (StefanRijnhart) [inline account_banking_nl_rabo_mt940/account_banking_nl_rabo_mt940.py]: Sorry, my mistake. Then there are no problems with uniqueness indeed.

However, does this mean that you are taking the account number out of the statement identifier (i.e. its display name)? Isn't that going to make it a little bit difficult for the user to quickly recognize statements for a particular account?

**Comment by pedrobaeza** [comment]
This is conflicting, so it seems there has been other improvements that makes this not needed. I'm closing this, but if this is still useful, feel free to reopen fixing conflicts.

**Pull request closed without merging**